### PR TITLE
Preload users instead of querying separately when searching for comments

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Optimization

## Description
The chase continues...somehow comments search, seemingly at random, throws a `500` because it can't find the user for the comment. This PR changes the logic to 1) preload users instead of querying for them separately and 2) add a conditional to the serializer to prevent an error and return an empty hash instead.

## Related Tickets & Documents
https://github.com/forem/forem/pull/13489

## QA Instructions, Screenshots, Recordings
Specs should cover us here, but you can pull this locally, enable the `search_2_comments` feature flag, and search for comments.

## Added tests?
- [x] No, and this is why: already have tests in place

## [Forem core team only] How will this change be communicated?
- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?
I'll sync with SRE to monitor the results of this change.

2_chainz_idk_gif

### CODE REVIEW

File: app/serializers/search/postgres_comment_serializer.rb [hunk 1]
```diff
@@ -40,9 +40,8 @@ class PostgresCommentSerializer < ApplicationSerializer
 
     # NOTE: not using the `NestedUserSerializer` to avoid hitting Redis to
     # fetch the cached value
-    attribute :user do |comment, params|
-      user = params[:users][comment.user_id]
-      user.slice(:name, :profile_image_90, :username).symbolize_keys
+    attribute :user do |comment|
+      comment.user.slice(:name, :profile_image_90, :username).symbolize_keys
```
Comment (rhymes) [inline app/serializers/search/postgres_comment_serializer.rb:44]: let's throw in a `if user` here too, as we do on line 7 and call it a day :D

Comment (atsmith813) [reply app/serializers/search/postgres_comment_serializer.rb]: Great call, updated 💯

**Review by rhymes — APPROVED** [review]
LGTM!

File: app/services/search/postgres/comment.rb [hunk 2]
```diff
@@ -55,6 +47,7 @@ def self.search_documents(
         per_page = [(per_page || DEFAULT_PER_PAGE).to_i, MAX_PER_PAGE].min
 
         relation = ::Comment
+          .includes(:user)
```
Comment (djuber) [inline app/services/search/postgres/comment.rb:50]: IIRC the idea originally behind not including user was the size of the objects (many columns in this table). 

Just reminding myself of tradeoffs we're making, not questioning your choice :)

**Review by djuber — APPROVED** [review]

**Pull request merged**